bob-backend/src/pearl/holder.rs

```diff
@@ -25,7 +25,7 @@ pub struct Holder {
     disk_path: PathBuf,
     config: PearlConfig,
     storage: Arc<RwLock<PearlSync>>,
-    last_write_ts: Arc<RwLock<u64>>,
+    last_modification: Arc<RwLock<u64>>,
```

Comment:
`RwLock` here protects only a variable, so there is no need to use async version of it. I suggest to use sync `RwLock` from `parking_lot` crate (https://docs.rs/parking_lot/latest/parking_lot/type.RwLock.html). It will be more efficient